task/buildah/0.1/buildah.yaml

```diff
@@ -105,6 +111,14 @@ spec:
     - name: YUM_REPOS_D_TARGET
       value: $(params.YUM_REPOS_D_TARGET)
   steps:
+  - name: use-trusted-artifact
+    image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:7a6a70bb972dba913e3d087305eece1a3e0462fd5adae52cd32322b9997bf213
+    args:
+      - use
+      - --store
+      - $(workspaces.source.path)
+      - $(params.SOURCE_ARTIFACT)=$(workspaces.source.path)/source
+      - $(params.CACHI2_ARTIFACT)=$(workspaces.source.path)/cachi2
```

Comment:
What happens if a user makes some modifications in the source directory in a custom task but doesn't use the trusted artifact mechanisms? I guess this step just throws away their modifications?

Would it be possible to warn the user or even fail the build when trusted-artifacts discards user modifications?

Reply:
A couple of things may happen:

1. The user adds their custom task to the chain of tasks where it produces a new trusted artifact which is then wired up to the buildah task. In this case, the build will succeed but EC will catch this as a violation because an untrusted task modified the source.
2. The user adds a custom task but doesn't wire things up properly in the pipeline. Then, yes, the changes will just get thrown out.
3. The user adds a custom task that tries to modify the trusted artifact produced by a trusted task. This will cause a build failure when another task tries to use that archive.